c10/util/string_view.h

```diff
@@ -591,6 +591,21 @@ constexpr inline void swap(
 
 using string_view = basic_string_view<char>;
 
+// NOTE: In C++20, this function should be replaced by str.starts_with
+constexpr bool string_view_starts_with(
+    std::string_view str,
+    std::string_view prefix) noexcept {
+  return str.find(prefix) == 0;
+}
+
+// NOTE: In C++20, this function should be replaced by str.ends_with
+constexpr bool string_view_ends_with(
+    std::string_view str,
+    std::string_view suffix) noexcept {
+  return (suffix.size() > str.size())
+      ? false
+      : (str.rfind(suffix) == (str.size() - suffix.size()));
```

Comment:
This could be a linear length search in regards to the length of string, unlike the code in you previous commit. It's less efficient.

Reply:
Use rfind, find is not optimal in both cases, we need to revert them to the old code.